src/receiver.ts

```diff
@@ -0,0 +1,103 @@
+import crypto from 'crypto';
+import tsscmp from 'tsscmp';
+import querystring from 'querystring';
+import App from './App';
+import { AckFn, AnyMiddlewareArgs } from './types';
+import { ReceiverAuthenticityError } from './errors';
+
+export interface ReceiverEvent {
+  body: Record<string, any>;
+  // TODO: there should maybe be some more help for implementors of Receiver to know what kind of argument the AckFn
+  // is expected to deal with.
+  ack: AckFn<any>;
+}
+
+export interface Receiver {
+  init(app: App): void;
+  start(...args: any[]): Promise<unknown>;
+  stop(...args: any[]): Promise<unknown>;
+}
+
+export function verifyRequestSignature(
```

Comment:
`parseRequestBody`, `verifyRequestSignature`, and `verifySignatureAndParseBody` will all eventually be moved into a new package